Neos.Flow/Classes/Mvc/Routing/RouteConfiguration.php

```diff
@@ -0,0 +1,120 @@
+     */
+    protected function createRoutesFromConfiguration()
+    {
+        if ($this->routesCreated === true) {
+            return;
+        }
+        $this->initializeRoutesConfiguration();
+        $routes = [];
+        $routesWithHttpMethodConstraints = [];
+        foreach ($this->routesConfiguration as $routeConfiguration) {
+            $route = new Route();
+            if (isset($routeConfiguration['name'])) {
+                $route->setName($routeConfiguration['name']);
+            }
+            $uriPattern = $routeConfiguration['uriPattern'];
+            $route->setUriPattern($uriPattern);
+            if (isset($routeConfiguration['defaults'])) {
+                $route->setDefaults($routeConfiguration['defaults']);
+            }
+            if (isset($routeConfiguration['routeParts'])) {
+                $route->setRoutePartsConfiguration($routeConfiguration['routeParts']);
+            }
+            if (isset($routeConfiguration['toLowerCase'])) {
+                $route->setLowerCase($routeConfiguration['toLowerCase']);
+            }
+            if (isset($routeConfiguration['appendExceedingArguments'])) {
+                $route->setAppendExceedingArguments($routeConfiguration['appendExceedingArguments']);
+            }
+            if (isset($routeConfiguration['httpMethods'])) {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === false) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678427);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = true;
+                $route->setHttpMethods($routeConfiguration['httpMethods']);
+            } else {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === true) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678432);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = false;
+            }
+            $routes = $route;
+        }
+        $this->routes = Routes::fromArray($routes);
+        $this->routesCreated = true;
+    }
+
+    /**
+     * Checks if a routes configuration was set and otherwise loads the configuration from the configuration manager.
+     *
+     * @return void
+     */
+    protected function initializeRoutesConfiguration()
+    {
+        if ($this->routesConfiguration === null) {
+            $this->routesConfiguration = $this->configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_ROUTES);
+        }
+    }
```

Comment:
and remove this :)

Reply:
And add a signal/slot, so we can integrate annotated routes and other external route configurations nicely 🤩

Reply:
I would suggest not to make this very concise class an extension point. Instead you should be able to replace it easily if you need a different config source